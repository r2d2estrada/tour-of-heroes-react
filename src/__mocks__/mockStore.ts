import configureStore from "redux-mock-store";
import { State } from "../redux";

const initialState: State = {
  heroes: [],
  selectedHero: null,
  messages: [],
};

const middlewares: any = [];

const createMockStore = configureStore(middlewares);

export const mockStore = createMockStore(initialState);
