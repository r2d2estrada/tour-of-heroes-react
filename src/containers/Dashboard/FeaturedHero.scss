.featured-hero {
  cursor: pointer;
  &:hover {
    background-color: var(--primary);
    color: var(--white);
  }
}
