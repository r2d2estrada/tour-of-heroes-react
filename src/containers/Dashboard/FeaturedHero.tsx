import React from "react";
import { Hero } from "../../models/hero";
import "./FeaturedHero.scss";

interface FeaturedHeroProps {
  hero: Hero;
}

const FeaturedHero: React.FC<FeaturedHeroProps> = ({ hero }) => {
  return (
    <div className="card featured-hero">
      <div className="card-body">
        <h2 className="card-title text-center">{hero.name}</h2>
      </div>
    </div>
  );
};

export default FeaturedHero;
