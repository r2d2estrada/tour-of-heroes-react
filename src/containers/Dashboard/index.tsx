import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { State } from "../../redux";
import { getHeroesAction } from "../../redux/actions/hero.actions";
import { Hero } from "../../models/hero";
import { Observable } from "rxjs";
import FeaturedHero from "./FeaturedHero";
import Messages from "../../components/Messages";
import { useDocumentTitle } from "../../utils/hooks";
import HeroSearch from "../../components/HeroSearch";

interface DashboardProps {
  heroes: Hero[] | any;
  getHeroes: Observable<Hero[]> | any;
}

export const Dashboard: React.FC<DashboardProps> = ({ heroes, getHeroes }) => {
  const [featuredHeroes, setFeaturedHeroes] = useState([]);

  useDocumentTitle("Dashboard");

  useEffect(() => {
    if (heroes.length === 0) getHeroes();
    else {
      setFeaturedHeroes(heroes.slice(1, 5));
    }
  }, [getHeroes, heroes]);

  return (
    <div className="Dashboard">
      <div className="row">
        <div className="col-md-12">
          <h2>Featured Heroes</h2>
        </div>
      </div>
      <hr />
      <div className="row">
        {featuredHeroes.map((hero: Hero) => (
          <div key={hero.id} className="col-md-3">
            <FeaturedHero hero={hero} />
          </div>
        ))}
      </div>
      <div className="mt-3">
        <HeroSearch heroes={heroes} />
      </div>
      <div className="mt-3">
        <Messages />
      </div>
    </div>
  );
};

const mapStateToProps = (state: State) => ({
  heroes: state.heroes,
});

const mapDispatchToProps = {
  getHeroes: getHeroesAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
