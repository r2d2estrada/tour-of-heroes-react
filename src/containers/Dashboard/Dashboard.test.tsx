import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Dashboard from ".";
import { mockStore } from "../../__mocks__/mockStore";

jest.mock("../../redux/actions/hero.actions");

describe("Dashboard", () => {
  let cpnt: any;

  beforeEach(() => {
    console.warn(mockStore.getState());
    cpnt = renderer.create(
      <Provider store={mockStore}>
        <Dashboard />
      </Provider>
    );
  });

  it("should match snapshot", () => {
    expect(cpnt).toMatchSnapshot();
  });
});
