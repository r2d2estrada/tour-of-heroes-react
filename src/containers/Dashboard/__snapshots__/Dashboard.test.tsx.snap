// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match snapshot 1`] = `
<div
  className="Dashboard"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h2>
        Featured Heroes
      </h2>
    </div>
  </div>
  <hr />
  <div
    className="row"
  />
  <div
    className="mt-3"
  >
    <div
      className="card hero-search"
    >
      <div
        className="card-header"
      >
        <h4>
          HeroSearch
        </h4>
      </div>
      <div
        className="card-body"
      >
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            aria-expanded={false}
            autoComplete="off"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="combobox"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-3"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h4>
              Messages:
            </h4>
          </div>
          <div
            className="col text-right"
          >
            <button
              className="btn btn-outline-secondary"
              onClick={[Function]}
            >
              Clear
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <ul
          className="list-group"
        />
      </div>
    </div>
  </div>
</div>
`;
